code/controllers/ContentController.php

```diff
@@ -216,9 +216,13 @@ public function handleRequest(SS_HTTPRequest $request, DataModel $model = null)
 	public function httpError($code, $message = null) {
 		// Don't use the HTML response for media requests
 		$response = $this->request->isMedia() ? null : ErrorPage::response_for($code);
+		// if we have a broken error and response is null do not reveal controller info
+		if (!$response && ($code < 200 || $code > 302)) {
+			$response = ErrorPage::response_for($code);
```

Comment:
This doesn't prevent an error page being returned if the response is a media type. If you have an <img src=brokenlinke.jpg> it should not attempt to return an ErrorPage. That's what the above condition attempts to prevent.